worker/uniter/runner/context/context.go

```diff
@@ -1882,6 +1888,30 @@ func (ctx *HookContext) WorkloadName() (string, error) {
 	return ctx.workloadName, nil
 }
 
+// WorkloadNoticeType returns the type of the notice for workload notice hooks.
+func (ctx *HookContext) WorkloadNoticeType() (string, error) {
```

Comment:
It looks like these new functions are only used by tests. Do they need to live here?

Reply:
I was copying where the other similar fields were exposed, but, no, they aren't used outside of the tests so could be moved (into one of the _test files I assume?). I might need some help with that, though, I'm not sure how to provide access in contextfactory_test to the fields that are in context. It's the same package but I get errors if I try to access it from contextfactory. I can try to figure it out Monday, but might need some hints.

Reply:
Okay, I see -- if the other similar fields are exposed this way, that's probably fine. Yeah, I see some of the others exposed in a similar way are only used by tests. The alternative would be to put them on `HookContext` in `export_test.go` -- which is in `package context` but has a `_test.go` filename so is compiled only when compiling tests -- see `HookContext.PendingSecretRemoves`, for example. But in this case I think you're right, precedent says to put them right where you have them.